plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -204,7 +204,7 @@ func findNodesThatFit(
 		workqueue.Parallelize(16, len(nodes), checkNode)
 		filtered = filtered[:filteredLen]
 		if len(errs) > 0 {
-			return []*v1.Node{}, FailedPredicateMap{}, errors.NewAggregate(errs)
+			return []*v1.Node{}, FailedPredicateMap{}, errors.UniqueAggregateWithCount(errors.NewAggregate(errs))
```

Comment:
I have separated functionality in 2 functions: one that calls flatten and one doesn't where it is known that the input is flattened already.